core/runtime/src/main/java/io/quarkus/runtime/ExecutorRecorder.java

```diff
@@ -159,9 +161,28 @@ private static EnhancedQueueExecutor createExecutor(ThreadPoolConfig threadPoolC
         }
         builder.setGrowthResistance(threadPoolConfig.growthResistance);
         builder.setKeepAliveTime(threadPoolConfig.keepAliveTime);
+
+        if (contextHandler != null) {
```

Comment:
...since you already `null`-check it here.

Reply:
I think I need coffee, not sure what you mean here.

If the build step now can pass a null in, do you mean to leave this check "as is"?

Reply:
Yeah this check would remain because either the build item would be optional (this would be preferred) or it can carry `null` (not as "nice" but might do in a pinch).